src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.EVP.MacAlgs.cs

```diff
@@ -0,0 +1,42 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using Microsoft.Win32.SafeHandles;
+
+internal static partial class Interop
+{
+    internal static partial class Crypto
+    {
+        internal static partial class EvpMacAlgs
+        {
+            internal static SafeEvpMacHandle? Kmac128 { get; } = EvpMacFetch(HashAlgorithmNames.KMAC128);
+            internal static SafeEvpMacHandle? Kmac256 { get; } = EvpMacFetch(HashAlgorithmNames.KMAC256);
+
+            [LibraryImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_EvpMacFetch", StringMarshalling = StringMarshalling.Utf8)]
+            private static partial SafeEvpMacHandle CryptoNative_EvpMacFetch(string algorithm, out int haveFeature);
+
+            private static SafeEvpMacHandle? EvpMacFetch(string algorithm)
+            {
+                SafeEvpMacHandle mac = CryptoNative_EvpMacFetch(algorithm, out int haveFeature);
+
+                if (haveFeature == 0)
+                {
+                    Debug.Assert(mac.IsInvalid);
+                    mac.Dispose();
+                    return null;
+                }
+
+                if (mac.IsInvalid)
+                {
+                    mac.Dispose();
+                    throw CreateOpenSslCryptographicException();
```

Comment:
OK, it looks like the native side is returning two things, 1) the EVP_MAC*, 2) whether or not the library even had the EVP_MAC_fetch function.

If the library doesn't have EVP_MAC_fetch then it doesn't have KMAC, so we'll return null and know there's no KMAC.  But if it does have EVP_MAC_fetch we expect the algorithm to be present, and if not, we throw.

Isn't that a problem if a) some distro or private build simply chose to build with KMAC disabled; or b) we try to use this routine for some algorithm introduced in the future?